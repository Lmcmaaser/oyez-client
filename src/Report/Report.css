/*mobile view*/
h3 {
  display: flex;
  text-align: center;
  margin: 0px 0px 10px 0px;
  font-size: 22px;
}

.form_group {
  display: flex;
  flex-direction: column;
  font-size: 18px;
  padding: 10px;
}

fieldset {
  display: flex;
  flex-direction: column;
  font-size: 20px;
  background-color: white;
  border: none;
  padding: 0px 5px 0px 5px;
  margin: 0px;
}

legend {
  display: none;
}

.main-label {
  font-weight: bold;
  margin: 5px 5px 0px 5px;
  align-self: center;
}

label {
  margin-bottom: 5px;
}

select {
  margin: 5px;
  padding: 5px;
  width: 60%;
  height: 30px;
  align-self: center;
}

input[type=text] {
  margin: 5px;
  padding: 5px;
  width: 50%;
  height: 30px;
  align-self: center;
}

input[type=radio] {
  margin: 5px;
  padding: 5px;
  height: 20px;
  width: 20px;
}

input[type=date] {
  margin: 5px;
  padding: 5px;
  width: 50%;
  height: 30px;
}

button.submit-button {
  display: flex;
  align-self: center;
  justify-content: center;
  font-size: 18px;
  font-weight: bold;
  padding: 7px;
  margin: 5px 10px 0px 10px;
  background-color: #B0AFE3;
  border: 5px solid #B0AFE3;
  border-radius: 8px;
}

/*desktop view*/
@media screen and (min-width: 600px) {
  h3 {
    margin: 10px 0px 10px 0px;
    font-size: 28px;
  }

  fieldset {
  display: flex;
  flex-direction: column;
  font-size: 22px;
  background-color: white;
  border: none;
  padding: 7px 5px 7px 5px;
  margin: 0px;
  }

  .main-label {
    font-weight: bold;
    margin: 5px 5px 0px 5px;
    align-self: center;
  }

  input[type=text] {
    margin: 5px;
    padding: 5px;
    width: 40%;
    height: 35px;
  }

  select {
    margin: 5px;
    padding: 5px;
    width: 40%;
    height: 35px;
  }
}
