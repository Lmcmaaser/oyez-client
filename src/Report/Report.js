import React from 'react';
import ValidationError from '../ValidationError.js';
import ApiContext from '../ApiContext';
import './Report.css';

export default class Report extends React.Component {
  static contextType = ApiContext;

  constructor(props) {
    super(props)
    this.state = {
      stateid: {
        value: '',
        touched: true
      },
      code: {
        value: '',
        touched: false
      },
      diagnosis_type: {
        value: '',
        touched: false
      },
      diagnosis_date: {
        value: '',
        touched: false
      },
      household: {
        value: '',
        touched: false
      }
    }
  }

  updateStateId(stateid) {
    console.log(stateid);
    this.setState({stateid: {value: stateid, touched: true}});
  }
  updateZipCode(code) {
    console.log(code);
    this.setState({code: {value: code, touched: true}});
  }
  updateDiagnosisType(diagnosis_type) {
    console.log(diagnosis_type);
    this.setState({diagnosis_type: {value: diagnosis_type, touched: true}});
  }
  updateDiagnosisDate(diagnosis_date) {
    console.log(diagnosis_date);
    this.setState({diagnosis_date: {value: diagnosis_date, touched: true}});
  }
  updateHousehold(household) {
    this.setState({household: {value: household, touched: true}});
  }

  handleSubmit(event) {
    event.preventDefault();
    const { code, diagnosis_type, diagnosis_date, household, stateid } = this.state;
    const report = {
      code: code.value,
      diagnosis_type: diagnosis_type.value,
      diagnosis_date: diagnosis_date.value,
      household: household.value,
      stateid: stateid.value
    }
    console.log(report);
    this.context.addReport(report);
    this.props.history.push('/submitted');
  }

  validateZipCode() {
    const zip_code = this.state.code.value.trim();
    if (zip_code.length > 5) {
      return "Zip code can only be 5 digits long"
    } else if (!zip_code.match(/[0-9]/)) {
      return "Zip code may only contain numbers"
    }
  }
  validateHousehold() {
    const household = this.state.household.value.trim();
    if (!household.match(/[0-9]/)) {
      return "Your response may only contain a numbers"
    }
  }

  render() {
    const zipCodeError = this.validateZipCode();
    const householdError = this.validateHousehold();

    return (
      <div className="report-page">
        <h3>Submit a Report</h3>
        <form className="form-group" onSubmit={event => this.handleSubmit(event)}>
          <fieldset>
            <legend>Report Form</legend>
              <label className="main-label" htmlFor="stateid">What state do you live in?</label>
              <select
                name="stateid"
                required
                aria-label="select state"
                onChange={event => this.updateStateId(event.target.value)}
              >
                {this.context.us_states.map(us_state =>
                  <option key={us_state.stateid} value={us_state.stateid}>{us_state.name}</option>
                )}
              </select>
              <label className="main-label">What is your zip code?</label>
                <input
                  type="text"
                  required
                  name="code"
                  id="code"
                  aria-label="input zip code"
                  value={this.state.code.value}
                  onChange={event => this.updateZipCode(event.target.value)}
                />
                  {this.state.code.touched && (
                    <ValidationError message={zipCodeError} />
                  )}
              <label className="main-label" htmlFor="diagnosis_type">
                How were you diagnosed?
              </label>
                <label htmlFor="container">
                  <input
                    type="radio"
                    id="test"
                    name="diagnosis_type"
                    value="test"
                    aria-label="select test diagnosis type"
                    onChange={event => this.updateDiagnosisType(event.target.value)}
                  />
                  <span className="checkmark"></span>
                 Tested positive</label>
                <label htmlFor="container">
                  <input
                    type="radio"
                    id="doctor"
                    name="diagnosis_type"
                    value="doctor"
                    aria-label="select doctor diagnosis type"
                    onChange={event => this.updateDiagnosisType(event.target.value)}
                  />
                  <span className="checkmark"></span>
                Doctor's professional assessment (diagnosis made without a test)</label>
                <label htmlFor="container">
                  <input
                    type="radio"
                    id="self"
                    name="diagnosis_type"
                    value="self"
                    aria-label="select self diagnosis type"
                    onChange={event => this.updateDiagnosisType(event.target.value)}
                  />
                  <span className="checkmark"></span>
                Self-diagnosis based on my symptoms</label>
              <label className="main-label">When were you diagnosed? </label>
                <input
                  type="date"
                  required
                  name="diagnosis_date"
                  id="diagnosis_date"
                  aria-label="date of diagnosis"
                  value={this.state.diagnosis_date.value}
                  onChange={event => this.updateDiagnosisDate(event.target.value)}
                />
              <label className="main-label">
                How many people live with you in your household?
              </label>
                <input
                  type="text"
                  required
                  name="household"
                  id="household"
                  aria-label="number people in your hosuehold"
                  value={this.state.household.value}
                  onChange={event => this.updateHousehold(event.target.value)}
                />
                  {this.state.household.touched && (
                    <ValidationError message={householdError} />
                  )}

              <button
                type="submit"
                className="submit-button"
                aria-label="submit-button"
                disabled={
                this.validateZipCode() ||
                this.validateHousehold()
                }
              >
                Submit
              </button>
          </fieldset>
        </form>
      </div>
    )
  }
}
