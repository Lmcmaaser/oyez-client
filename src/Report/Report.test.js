import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Report from './Report'

it('renders a .report-page by default', () => {
  const wrapper = shallow(<Report />)
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders without crashing', () => {
  shallow(<Report/>);
});
