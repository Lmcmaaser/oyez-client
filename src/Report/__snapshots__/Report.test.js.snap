// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a .report-page by default 1`] = `
<div
  className="report-page"
>
  <h3>
    Submit a Report
  </h3>
  <form
    className="form-group"
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        Report Form
      </legend>
      <label
        className="main-label"
        htmlFor="stateid"
      >
        What state do you live in?
      </label>
      <select
        aria-label="select state"
        name="stateid"
        onChange={[Function]}
        required={true}
      />
      <label
        className="main-label"
      >
        What is your zip code?
      </label>
      <input
        aria-label="input zip code"
        id="code"
        name="code"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        className="main-label"
        htmlFor="diagnosis_type"
      >
        How were you diagnosed?
      </label>
      <label
        htmlFor="container"
      >
        <input
          aria-label="select test diagnosis type"
          id="test"
          name="diagnosis_type"
          onChange={[Function]}
          type="radio"
          value="test"
        />
        <span
          className="checkmark"
        />
        Tested positive
      </label>
      <label
        htmlFor="container"
      >
        <input
          aria-label="select doctor diagnosis type"
          id="doctor"
          name="diagnosis_type"
          onChange={[Function]}
          type="radio"
          value="doctor"
        />
        <span
          className="checkmark"
        />
        Doctor's professional assessment (diagnosis made without a test)
      </label>
      <label
        htmlFor="container"
      >
        <input
          aria-label="select self diagnosis type"
          id="self"
          name="diagnosis_type"
          onChange={[Function]}
          type="radio"
          value="self"
        />
        <span
          className="checkmark"
        />
        Self-diagnosis based on my symptoms
      </label>
      <label
        className="main-label"
      >
        When were you diagnosed? 
      </label>
      <input
        aria-label="date of diagnosis"
        id="diagnosis_date"
        name="diagnosis_date"
        onChange={[Function]}
        required={true}
        type="date"
        value=""
      />
      <label
        className="main-label"
      >
        How many people live with you in your household?
      </label>
      <input
        aria-label="number people in your hosuehold"
        id="household"
        name="household"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <button
        aria-label="submit-button"
        className="submit-button"
        disabled="Zip code may only contain numbers"
        type="submit"
      >
        Submit
      </button>
    </fieldset>
  </form>
</div>
`;
